File: api/account/urls.py
# account/urls.py

from django.urls import path
from . import views

urlpatterns = [
    path('user/login/', views.login_user, name="login"),
    path('user/signup/', views.register_user, name="register"),
    path('user/register/phone/', views.register_user_phone_number, name="register-phone"),
    path('user/logout/', views.logout_user, name="logout_user"),
    path('user/verify/phone/', views.verify_code, name="verify_code"),
    path('user/change-password/', views.change_user_password, name="change_password"),
    path('user/resend/code/', views.resend_phone_code, name="resend_code"),
    path('user/info/', views.get_user_info, name="user_info"),
    path('user/start-session/', views.StartUserSessionView.as_view(), name='start-session'),
    path('user/session-heartbeat/', views.SessionHeartbeatView.as_view(), name='session-heartbeat'),
    path('user/end-session/', views.EndUserSessionView.as_view(), name='end-session'),
    path('user/refresh-token/', views.refresh_access_token, name='refresh-token'),
]

File: src/services/utils/urls.ts
// src/services/urls.ts
export const BASE_URL_WITHOUT_TRANSPORT = '127.0.0.1:8000';
export const BASE_URL = `http://${BASE_URL_WITHOUT_TRANSPORT}`;
// 192.168.1.212 
// 127.0.0.1:8000
// API Endpoints
export const apiUrls = {
  // User account-related endpoints
  login: 'account/user/login/',
  signup: 'account/user/signup/',
  logout: 'account/user/logout/',
  registerPhone: 'account/user/register/phone/',
  verifyPhone: 'account/user/verify/phone/',
  changeUserPassword: 'account/user/change-password/',
  resetUserPassword: 'account/user/reset-password/',
  resendPhoneCode: 'account/user/resend/code/',
  userProfile: 'account/user/info/',
  refreshToken: 'account/user/refresh-token/',
  updateProfilePicture: 'account/user/update-profile-picture/',

  // Property-related endpoints
  addProperty: 'properties/add/',
  exploreNearbyProperties: 'properties/explore/near-by/',
  exploreInterestedProperties: 'properties/explore/interested/',
  getPropertyById: 'properties/property/',
  saveProperty: 'properties/:id/save/',
  getUserProperties: 'properties/user/my-properties/',
  getSavedProperties: 'properties/user/saved-properties/',
  makePropertyFeatured: 'properties/user/my-properties',

  // Session-related endpoints
  startSession: 'account/user/start-session/',
  sessionHeartbeat: 'account/user/session-heartbeat/',

  // Message-related endpoints
  fetchConversations: 'messaging/conversations/',
  fetchSentMessages: 'messaging/sent/',
  fetchChats: 'messaging/chats/',

};

// WebSocket URLs
export const getLogInteractionsWebSocketUrl = (): string =>
  `ws://${BASE_URL_WITHOUT_TRANSPORT}/ws/log-interactions/`;
export const getMessagesWebSocketUrl = (): string =>
  `ws://${BASE_URL_WITHOUT_TRANSPORT}/ws/messages/`;
